pil-analyzer/src/evaluator.rs

```diff
@@ -686,7 +689,7 @@ mod test {
     #[test]
     pub fn capturing() {
         let src = r#"namespace Main(16);
-            let f = |n, g| match n { 99 => |i| n, 1 => g(3) };
+            let f: int, int -> int = |n, g| match n { 99 => |i| n, 1 => g(3) };
```

Comment:
I don't understand the type of the RHS. Aren't the types of the match arms different? `T -> T` and `T`

Reply:
true, this does not type check. Will fix.